libegg/eggdesktopfile.c

```diff
@@ -1224,7 +1224,7 @@ egg_desktop_file_launchv (EggDesktopFile *desktop_file,
 
       startup_id = NULL;
 
-#ifdef GDK_WINDOWING_X11
+#if defined(GDK_WINDOWING_X11) || defined(HAVE_X11)
```

Comment:
[GDK_IS_X11_DISPLAY](https://github.com/GNOME/gtk/blob/mainline/gdk/x11/gdkx11display.h), [gdk_x11_screen_get_screen_number](https://github.com/GNOME/gtk/blob/mainline/gdk/x11/gdkx11screen.h) methods require GDK_WINDOWING_X11

mate-panel uses HAVE_X11 for other purposes:

```
# Check if we have the X development libraries
have_x11=no
if test "x$enable_x11" != "xno"; then
  PKG_CHECK_MODULES(X, x11 xau, have_x11=yes, [
      if test "x$enable_x11" = "xyes"; then
        AC_MSG_ERROR([X development libraries not found])
      fi
    ])
fi

AM_CONDITIONAL(ENABLE_X11, [test "x$have_x11" = "xyes"])

if test "x$have_x11" = "xyes"; then
  AC_DEFINE(HAVE_X11, 1, [Have the X11 development library])
fi
```

Reply:
Nope, i restored the original code.
https://github.com/mate-desktop/mate-panel/commit/1cfcb58dd5bc3e6effac1d2cdbc482dce930dd9b#diff-4f59a5256ba9315f26f013d5ebad5476L1215
here in eggdesktopfile.c and eggsmclient-xsmp.c `HAVE_X11` is used all over the place.

Reply:
And this is a `OR` condition.

Reply:
PR https://github.com/mate-desktop/mate-panel/pull/1099

Reply:
Congratulation, your PR is broken :)
You're simply wrong....

Reply:
I have only found those X11 libraries (mate-submodules), which are in LIBEGG (mate-panel):
```
mate-submodules $ git grep X11 | grep include
libegg/eggsmclient-xsmp.c:#include <X11/SM/SMlib.h>
libegg/eggsmclient-xsmp.c:#include <X11/ICE/ICElib.h>
```
```
mate-panel $ grep LIBEGG configure.ac 
PKG_CHECK_MODULES(LIBEGG, ice sm gtk+-3.0)
```

Reply:
To clarify, `HAVE_X11` **guarantees** `GDK_WINDOWING_X11` is true. If `HAVE_X11` is true ever true and `GDK_WINDOWING_X11` is not then the build system is broken. `HAVE_X11`/`HAVE_WAYLAND` are useful because it allows us to disable one platform or the other even if GTK supports it. We might want to do that for testing, or to build without our additional dependencies required for that platform.